src/AudioSystem.cpp

```diff
@@ -21,25 +36,42 @@ AudioSystem::AudioSystem() {
     }
 
     // Load all audio files
-    load_audio_assets();
+    init_successful_ = load_audio_assets();
+
+    if ( !init_successful_) {
+        std::cerr << "AudioSystem initialization failed" << std::endl;
+        return;
+    }
 
     add_event_handler(EventType::KEYPRESS_EVENT, &AudioSystem::handle_keypress_event, this);
 }
 
-void AudioSystem::load_audio_assets() {
+bool AudioSystem::load_audio_assets() {
     // Load all sound assets
-    sound_assets_["beat"] = Mix_LoadWAV("assets/audio/beat.wav");
-
-    if (sound_assets_.at("beat") == nullptr) {
-        std::cerr << "Failed to load sound: beat.wav" << std::endl;
+    for (auto& sound_asset_info : SOUND_ASSETS_INFO) {
+        const int key = (int) sound_asset_info.first;
+        const char* path = sound_asset_info.second;
+        sound_assets_[key] = Mix_LoadWAV(path);
+
+        if (sound_assets_.at(key) == nullptr) {
+            std::cerr << "Failed to load sound: " << key << std::endl;
+            return false;
```

Comment:
If we fail to load one sound asset here, then we won't try to load any other audio asset. Wouldn't it be better to try to be robust here and continue loading instead?